src/Npgsql/SqlQueryParser.cs

```diff
@@ -516,17 +506,21 @@ void MoveToNextBatchCommand()
     }
 
     static bool IsLetter(char ch)
-        => 'a' <= ch && ch <= 'z' || 'A' <= ch && ch <= 'Z';
+        // [a-zA-Z]
+        => (uint)((ch | 0x20) - 'a') <= ('z' - 'a');
```

Comment:
Could you please add a comment here to describe how it works (since it's not that easy to understand)?

Or, if that's just a copy-paste of [Char.IsAsciiLetter](https://github.com/dotnet/runtime/blob/60cfaec2e6cffeb9a006bec4b8908ffcf71ac5b4/src/libraries/System.Private.CoreLib/src/System/Char.cs#L236) from the runtime repo, a simple reference to it should suffice.

Reply:
I initially added it to string.Equals https://github.com/dotnet/coreclr/pull/9213 I see that it has now been extracted out to char but I'm happy to link to that as the source.